app/controllers/assessments_controller.rb

```diff
@@ -641,6 +642,7 @@ def releaseSectionGrades
     num_released = releaseMatchingGrades { |submission, _| @cud.CA_of? submission.course_user_datum }
 
     if num_released > 0
+      @course.update_course_no_submissions_watchlist_instances(@cud)
```

Comment:
The way I went for to distinguish between a section level and a course level update is whether there exists a cud parameter for the function call. If there is none, then it's the instructor releasing all grades. Otherwise, that grader cud information is passed all the way down to the WatchlistInstance function call, to filter out the students in the course that the grader is not in charge of.

Reply:
ah i see. Understood